var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -42,17 +46,44 @@ def install(self, spec, prefix):
         shutil.copytree('mod', prefix.mod)
         if spec.satisfies('+python'):
             copy_tree('resources/neurodamus-py', prefix.python)
-        if spec.satisfies('+common'):
-            copy_all('resources/common/hoc', prefix.hoc)
-            copy_all('resources/common/mod', prefix.mod)
 
         filter_file(r'UNKNOWN_CORE_VERSION', r'%s' % spec.version, prefix.hoc.join('defvar.hoc'))
-
         try:
             commit_hash = self.fetcher[0].get_commit()
             filter_file(r'UNKNOWN_CORE_HASH', r'\'%s\'' % commit_hash, prefix.hoc.join('defvar.hoc'))
         except Exception as e:
-            tty.warn(e)
+            tty.warn(str(e))
+
+        # +Common will bring common mods and build a bare nrnmechlib
+        if spec.satisfies('+common'):
+            copy_all('resources/common/hoc', prefix.hoc)
+            copy_all('resources/common/mod', prefix.mod)
+
+            with working_dir(prefix):
+                which('nrnivmodl')('-incflags', '-DDISABLE_REPORTINGLIB', 'mod')
+
+                bindir = os.path.basename(self.neuron_archdir)
+                special = join_path(bindir, 'special')
+                open('.bindir', 'w').write(bindir)
```

Comment:
Note: We have to dump somewhere the info of the bindir because inside setup_environment() we can't access a property of the dependency.

Reply:
Isn't [setup_dependent_environment](https://spack.readthedocs.io/en/latest/spack.html#spack.package.PackageBase.setup_dependent_environment) do the job?

Reply:
Even for this own package. If we request module regeneration `self.neuron_archdir` fails...

Reply:
@pramodk I don't remember if there was a better way of having this working. In either case I think it's not so bad to have a config file written down.